GitHubDashboard/Controllers/QueryController.cs

```diff
@@ -132,6 +132,39 @@ private async Task<IReadOnlyList<Issue>> GetIssuesAsync(string owner, string rep
             // Catch it in the calling function...
             var issues = await _gitHubClient.Issue.GetAllForRepository(owner, repository, issueRequest);
             issues = issues.Where(i => i.PullRequest == null).ToList();
+
+            // We now need to exclude the milestone
+            if (!String.IsNullOrEmpty(excludedMilestone) && !(excludedMilestone == "undefined"))
+            {
+                issues = issues.Where(i => i.Milestone == null || i.Milestone.Title != excludedMilestone).ToList();
+            }
+
+            // We now need to exclude all the issues that have labels that should be excluded
+            if (!String.IsNullOrEmpty(excludedLabels) && !(excludedLabels == "undefined"))
+            {
+                var filteredIssues = new List<Issue>();
+                var excludedLabelValues = excludedLabels.Split(',');               
+
+                foreach(Issue i in issues)
+                {
+                    bool skip = false;
+                    foreach(Label l in i.Labels)
+                    {
+                        if(excludedLabelValues.Contains(l.Name))
```

Comment:
Instead of use `skip` here, consider:

```C#
if (!excludedLabelValues.Contains(l.Name))
{
    filteredIssues.Add(i);
}
```

Reply:
That wouldn't work because you will be adding the same element several times. You will need a way to keep track that you only add it once to the list. 

---
In reply to: [247579730](https://github.com/Pilchie/GitHubDashboard/pull/13#discussion_r247579730) [](ancestors = 247579730)

Reply:
Then make it a `HashSet` instead of a list?

Reply:
@jjmew explained to me on IM that I wasn't really reading the whole thing.

Could be replaced with `issues = issues.Where(i => i.Labels.Intersect(excludedLabelValues).Count == 0);`, but that would generate a bunch of garbage for the temporary interections.